src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,236 @@
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(GET_ALL_QUERY)) {
+            ResultSet resultSet = statement.executeQuery();
+            Map<Car, List<Driver>> driversMapByCar = new HashMap<>();
+            while (resultSet.next()) {
+                fillMap(driversMapByCar, resultSet, isDriverId(resultSet));
+            }
+            return retrieveCarsFromMap(driversMapByCar);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from DB.", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update car in DB. Car: " + car, e);
+        }
+        deleteDriversByCarInDB(car);
+        insertDriversByCarInDB(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car by id=" + id + " in DB.", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String carsByDriverQuery = GET_ALL_QUERY + " AND drivers.id = ?";
+        Map<Car, List<Driver>> driversMapByCar = new HashMap<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(carsByDriverQuery)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                fillMap(driversMapByCar, resultSet, true);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Couldn't get a list of cars by driver id=" + driverId + " from DB.", e);
+        }
+        String driversByCarQuery = GET_ALL_QUERY + " AND cars.id = ? AND drivers.id != ?";
```

Comment:
looks really strange, lets get all info about cars(without drivers), and then get only drivers for these cars